ludwig/models/predictor.py

```diff
@@ -130,7 +130,13 @@ def batch_predict(
         for key, pred_value_list in predictions.items():
             predictions[key] = tf.concat(pred_value_list, axis=0).numpy()
 
-        return from_numpy_dataset(predictions)
+        predictions = from_numpy_dataset(predictions)
+
+        for of_name, output_feature in model.output_features.items():
```

Comment:
We should probably pull this into `ray.py` as otherwise we will flatten but never unflatten for other backends.

Reply:
Good catch! Fixed in the latest commit.